packages/core-data/src/resolvers.js

```diff
@@ -78,3 +82,19 @@ export async function* getThemeSupports() {
 	const index = await apiRequest( { path: '/' } );
 	yield receiveThemeSupportsFromIndex( index );
 }
+
+/**
+ * Requests Upload Permissions from the REST API.
+ */
+export async function* hasUploadPermissions() {
+	const header = await apiRequest( { path: '/wp/v2/media', method: 'OPTIONS' } )
+		.then( ( body, status, xhr ) => {
+			if ( body.headers ) {
+				return body.headers.Allow;
+			}
+
+			return xhr.getResponseHeader( 'allow' );
```

Comment:
In what circumstance would this condition be satisfied? I cannot tell by the unit tests.

Reply:
@imath - I might need your help to answer this question. I'm guessing it's to satisfy particular browsers (IE)?

Is the fact that there's an xhr param to the response part of a shim we're using for fetch?

Reply:
@tallsan I must check in details but I think what you get with preloaded path is different than what you get with a regular REST request, so I probably added this if for some reason the path is not preloaded.

Reply:
Yes i confirm that was the reason. I did that when api-request.js, now api-fetch is used maybe it's no more needed.

Reply:
@talldan FYI to adapt to apiFetch and be sure to get the `allow` response header if the preloaded path is not available, here's how i've adapted this part:

```patch
diff --git packages/core-data/src/resolvers.js packages/core-data/src/resolvers.js
index 104106808..7836a1aec 100644
--- packages/core-data/src/resolvers.js
+++ packages/core-data/src/resolvers.js
@@ -116,13 +116,17 @@ export async function* getEmbedPreview( state, url ) {
  * Requests Upload Permissions from the REST API.
  */
 export async function* hasUploadPermissions() {
-	const header = await apiFetch( { path: '/wp/v2/media', method: 'OPTIONS' } )
-		.then( ( body, status, xhr ) => {
-			if ( body.headers ) {
-				return body.headers.Allow;
+	const header = await apiFetch( { path: '/wp/v2/media', method: 'OPTIONS', parse: false } )
+		.then( ( response ) => {
+			if ( ! response.headers.Allow ) {
+				for ( var i of response.headers.entries() ) {
+					if ( 'allow' === i[0] ) {
+						response.headers.Allow = i[1];
+					}
+				}
 			}
 
-			return xhr.getResponseHeader( 'allow' );
+			return response.headers.Allow;
 		} );
 
 	yield receiveUploadPermissions( includes( header, 'POST' ) );
```

Reply:
Looks like you can use headers.get, here's what I ended up with:
```
/**
 * Requests Upload Permissions from the REST API.
 */
export async function* hasUploadPermissions() {
	const response = await apiFetch( { path: '/wp/v2/media', method: 'OPTIONS', parse: false } );

	let allowHeader;
	if ( hasIn( response, [ 'headers', 'get' ] ) ) {
		// If the request is fetched using the fetch api, the header can be
		// retrieved using the 'get' method.
		allowHeader = response.headers.get( 'allow' );
	} else {
		// If the request was preloaded server-side and is returned by the
		// preloading middleware, the header will be a simple property.
		allowHeader = get( response, [ 'headers', 'Allow' ], '' );
	}
	yield receiveUploadPermissions( includes( allowHeader, 'POST' ) );
}
```

Reply:
Awesome work 💪👏